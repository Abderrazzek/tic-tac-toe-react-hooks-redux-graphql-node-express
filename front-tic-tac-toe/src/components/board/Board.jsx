import React from "react";

import Square from "../square/Square";

const Board = ({ squares, onClick }) => (
  <div data-testid="board" className="board">
    {squares.map((square, i) => (
      <Square key={i} value={square} onClick={() => onClick(i)} />
    ))}
  </div>
);

export default Board;
