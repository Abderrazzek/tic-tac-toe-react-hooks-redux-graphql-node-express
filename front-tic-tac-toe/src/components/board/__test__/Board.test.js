import { cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import ReactDom from "react-dom";
import React from "react";

import Board from "../Board";

//cleanup after each element rendering
afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDom.render(
    <Board squares={["", "X", "O", "", "X", "X", "O", ""]} />,
    div
  );
});

it("matches snapshot", () => {
  const component = renderer.create(
    <Board squares={["", "X", "O", "", "X", "X", "O", ""]} onClick={() => {}} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
