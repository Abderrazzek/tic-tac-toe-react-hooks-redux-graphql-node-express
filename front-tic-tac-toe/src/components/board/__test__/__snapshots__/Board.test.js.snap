// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="board"
  data-testid="board"
>
  <button
    className="squares"
    data-testid="square"
    onClick={[Function]}
  >
    
  </button>
  <button
    className="squares X"
    data-testid="square"
    onClick={[Function]}
  >
    X
  </button>
  <button
    className="squares O"
    data-testid="square"
    onClick={[Function]}
  >
    O
  </button>
  <button
    className="squares"
    data-testid="square"
    onClick={[Function]}
  >
    
  </button>
  <button
    className="squares X"
    data-testid="square"
    onClick={[Function]}
  >
    X
  </button>
  <button
    className="squares X"
    data-testid="square"
    onClick={[Function]}
  >
    X
  </button>
  <button
    className="squares O"
    data-testid="square"
    onClick={[Function]}
  >
    O
  </button>
  <button
    className="squares"
    data-testid="square"
    onClick={[Function]}
  >
    
  </button>
</div>
`;
